specifications/transactions/offchain-api/README.md

```diff
@@ -0,0 +1,256 @@
+
+`invalid_http_header`:
+
+* `X-REQUEST-SENDER-ADDRESS` header value is not the request sender’s address in the command object. All command objects should have a field that is the request sender’s address. For payment objects, it is sender.address or receiver.address.
+* Could not find Diem’s onchain account by the `X-REQUEST-SENDER-ADDRESS` header value.
+* Could not find the compliance key of the onchain account found by the `X-REQUEST-SENDER-ADDRESS` header value.
+* The compliance key found from the onchain account by `X-REQUEST-SENDER-ADDRESS` is not a valid ED25519 public key.
+* `X-REQUEST-ID` is not in valid UUID format.
+
+`missing_http_header`: missing HTTP header `X-REQUEST-ID` or `X-REQUEST-SENDER-ADDRESS`.
+
+####### JWS Validation Error Codes
+
+`invalid_jws`: invalid JWS format (compact) or protected header
+
+`invalid_jws_signature`: JWS signature verification failed
+
+####### Request Object Validation Error Codes
+
+`invalid_json`: decoded JWS body is not json.
+
+`invalid_object`: command request/response object json is not object, or the command object type does not match `command_type`.
+
+`missing_field`: A required field contains no data or is missing.
+
+`unknown_field`: field is unknown for an object.
+
+`unknown_command_type`: invalid/unsupported `command_type`.
+
+`invalid_field_value`: A field does not match the expected type, e.g., a UUID is not correctly formatted or an enum field contains an invalid value.
+
+`invalid_recipient_signature`:
+* Field `recipient_signature` value is not hex-encoded bytes.
+* Field `recipient_signature` value is an invalid signature.
+
+## Network Error Handling
+
+In the case of network failure, the sending party for a Command is expected to re-send the Command until it gets a response from the counterparty VASP. An exponential backoff is suggested for Command re-sends.
+
+Upon receipt of a Command that has already been processed (resulting in a success response or a Command error), the receiving side must reply with the same response as was previously issued (successful commands, or commands that fail with a command error, are idempotent). Requests that resulted in protocol errors may result in different responses.
+
+For example:
+* VASP A sends a request to VASP B which failed with an unknown network error. VASP A retries the send request to VASP B.
+* VASP B receives both requests, and processes them in parallel.
+* Both requests attempt to acquire a common lock. The first request that VASP A believes failed acquires the lock and the second request fails.
+* VASP B successfully processes the first request and sends back a successful response, however, VASP A doesn’t receive the response due to the previous errors.
+* VASP B rejects the second request due to lock contention.
+* VASP A receives the failure for the second request, and later, retries the same request. If VASP B processes the request when the lock is not under contention it will respond successfully.
+
```

Comment:
we need mention x-request-uuid or cid is used for idempotent key in above case. 
it is clear x-request-uuid should be regenerated when retry a request, but it is not clear what we should do with `cid` during retry.
also, there are 2 types retries:
1. within same process/thread, an http client retries the http request. For this case, I think `cid` must be same, but x-request-uuid is not clearly defined. In practice, updating `x-request-uuid` during http client retry requires extra work, and likely not be implemented if no specific call out.  I think it's fine to keep `x-request-uuid` same, which means x-request-uuid is idempotent key.
2. after call failed, application put the request into a retry queue, and retry later,  `x-request-uuid` should be different, we may define whether `cid` is same or different, either way I feel it does not matter anymore, as `x-request-uuid` becomes idempotent key because of above point, `cid` is redundant.

Reply:
good call out